src/liberty/devCommands.ts

```diff
@@ -102,6 +102,121 @@ export async function startDevMode(libProject?: LibertyProject | undefined): Pro
     }
 }
 
+
+export async function removeProject(uri: vscode.Uri): Promise<void> {
+    const projectProvider: ProjectProvider = ProjectProvider.getInstance();
+    if (undefined !== uri && undefined !== uri.fsPath ) {
+        // Right mouse clicked on a root folder, or on empty space with only one folder in workspace.
+        // Add project if:
+        // 1. Project has build files (pom.xml or build.gradle)
+        // 2. Not in liberty dashboard
+        // Once added, presist the data in workspace storage.\
+        // check if the path is in current list
+        const file = projectProvider.isPathExistsInPersistedProjects(uri.fsPath);
+        if (undefined !== file) {
+            const yes = localize("confirmation.button.label.yes");
+            const no = localize("confirmation.button.label.no");
+            vscode.window
+                .showInformationMessage(localize("remove.custom.project.confirmation", uri.fsPath), yes, no)
+                .then(answer => {
+                    if (answer === yes) {
+                    // delete and save
+                        projectProvider.removeInPersistedProjects(file);
+                        vscode.window
+                            .showInformationMessage(localize("remove.custom.project.successful"));
+                            projectProvider.fireChangeEvent();
+                    }
+                });
+        } else {
+            const message = localize("remove.custom.project.not.in.list");
```

Comment:
Even if not manually added to the dashboard, users should be able to remove automatically added projects from the dashboard too.

Reply:
The project will be automatically added back when the user re-opens the workspace, or make any changes to workspace, I don't see any benefit of doing this , plus this is also not covered by the design mentioned in #120.

Reply:
@travis1111 Is there a way to prevent the project from being automatically added back when the user re-opens the workspace? 

I believe this was the intention with the wording for https://github.com/OpenLiberty/liberty-tools-vscode/issues/120 "Allow for projects to be manually added/removed from the Liberty dashboard" - as in any project can be added or removed (and the addition/removal should persist) - at least that is how I understood this. I can see how this could be understood differently though - probably should have been more clear in the issue.

@kathrynkodama how much value loss is there if users cannot remove automatically added projects?

Reply:
Yes, sorry about that @travis1111, should have updated #120 for clarity after we finalized the current design. In the UFO design doc we do say "Projects can be manually added and removed to the dashboard view" and "Projects manually added and removed to the dashboard view will persist between VS Code sessions for the configured VS Code workspace". 

I can see a use case where we add a Liberty project automatically that they do not want in the dashboard view, especially not having the multi module story supported yet. In that event, it would be quite annoying to have the project automatically added each time they open that workspace.

Reply:
Opened https://github.com/OpenLiberty/liberty-tools-vscode/issues/142